sdx_controller/controllers/connection_controller.py

```diff
@@ -67,7 +67,9 @@ def getconnection_by_id(connection_id):
     :rtype: Connection
     """
     value = db_instance.read_from_db(f"{connection_id}")
-    return value
+    if not value:
+        return "Connection not found", 404
+    return json.loads(value[connection_id])
```

Comment:
Heads-up: converting from string to dict was necessary due to the API spec, which suggests a returned object of the type application/json: https://github.com/atlanticwave-sdx/sdx-controller/blob/main/sdx_controller/swagger/swagger.yaml#L181